lib/Query.js

```diff
@@ -753,6 +753,16 @@ class Query {
     return this.prepareSql(sql);
   }
 
+  async prepareTruncate(options) {
+    this.setOptions(options);
+
+    this.config.forTruncate = true;
+
+    const sql = this.sql.delete(this.getTable());
```

Comment:
Ideally, this method should do a `TRUNCATE` call, not a `DELETE` call. The main difference being that `TRUNCATE` would also reset the auto-increment sequences. But unfortunately, this is not supported by sql-bricks (i.e. `sql.truncate(tableName)` which makes this task slightly complicated

Reply:
Something like `this.sql(\`TRUNCATE TABLE ${this.getTable()} \`)` might be acceptable but we'd need to check the syntax in all SQL databases to verify that this is the standard.

Reply:
@joelmukuthu I wasn't aware that auto-increment sequences would be reset on Truncation and not on Deletion, it makes sense.

> this.sql(`TRUNCATE TABLE ${this.getTable()}`);

Also I am not sure if we can use above statement directly because for postgres there's no **TABLE** in the statement. It's just `TRUNCATE tableName`

Reply:
There is actually but it's optional &ndash; `TRUNCATE TABLE` seems to be what's supported by most databases. Anyway I'd like to do some more research before proceeding with this as this would be the first query Knorm is generating itself, so far all queries have been generated by sql-bricks. Also I could ask there if they're willing to add a `.truncate` method.

Alternatively, to lessen the scope we could add this first to @knorm/postgres and later move it to @knorm/knorm when other plugins are added.

Reply:
@joelmukuthu I was about to raise an issue with sql-bricks to ask for Truncate feature.

Reply:
Perfect! Go for it, I'll back you up :)

Reply:
@joelmukuthu I have added a feature request on [sql-bricks](https://github.com/CSNW/sql-bricks/issues/110).